src/RangeBase.ts

```diff
@@ -19,184 +23,256 @@ interface RangeBaseConfig extends BaseletConfig {
+                  rangeKey,
+                  bucketData,
+                  firstRangeOccurence.index,
+                  bucketData.length - 1,
+                  true
+                )
+                const targetIndex = getIndex(
+                  data[idKey],
+                  idKey,
+                  bucketData,
+                  firstRangeOccurence.index,
+                  lastRangeOccurence.index
+                )
+                if (targetIndex.found === true) {
+                  throw new Error(
+                    'Cannot insert data because id already exists'
+                  )
+                }
+                bucketData.splice(targetIndex.index, 0, data)
+              }
+              return disklet.setText(bucketPath, JSON.stringify(bucketData))
+            },
+            error => {
+              console.log(error)
+              console.log('assuming bucket doesnt exist')
+              // we must be sure bucket doesnt exist, else we'll overwrite
+              // we have to identify the error
+              return disklet.setText(bucketPath, JSON.stringify([data]))
+            }
+          )
+      },
+      query(
+        partition: string = '/',
+        rangeStart: number,
+        rangeEnd: number = rangeStart
+      ): Promise<any[]> {
+        const { bucketSize, rangeKey } = configData
+        const formattedPartition = checkAndformatPartition(partition)
+        // sanity check the range
+        const bucketFetchers = []
+        for (
+          let bucketNumber = Math.floor(rangeStart / bucketSize);
+          bucketNumber <= Math.floor(rangeEnd / bucketSize);
+          bucketNumber++
+        ) {
+          bucketFetchers.push(
+            disklet
+              .getText(
+                `${databaseName}${formattedPartition}/${bucketNumber}.json`
+              )
+              .then(rawBucketData => JSON.parse(rawBucketData))
+              .catch(error => {
+                console.log(
+                  `Error getting data from bucket ${bucketNumber}. ${error}`
+                )
+                return []
+              })
+          )
+        }
+        return Promise.all(bucketFetchers).then(bucketList => {
+          let queryResults: any[] = []
+          for (let i = 0; i < bucketList.length; i++) {
+            if (i === 0) {
+              const firstRangeIndex = getIndex(
+                rangeStart,
+                rangeKey,
+                bucketList[i]
+              )
+              queryResults = bucketList[i].slice(firstRangeIndex.index)
+            } else if (i === bucketList.length - 1) {
+              const lastRangeIndex = getIndex(
+                rangeEnd,
+                rangeKey,
+                bucketList[i],
+                0,
+                bucketList[i].length - 1,
+                true
+              )
+              Array.prototype.push.apply(
+                queryResults,
+                bucketList[i].slice(lastRangeIndex.index)
```

Comment:
This slice will return the latter half of the array. It looks like you actually want the first half of the array up until the last range index.